snuba/redis.py

```diff
@@ -31,24 +39,44 @@ def execute_command(self, *args: Any, **kwargs: Any) -> Any:
             return super(self.__class__, self).execute_command(*args, **kwargs)
 
 
-redis_client: RedisClientType
-
-if settings.USE_REDIS_CLUSTER:
-    startup_nodes = settings.REDIS_CLUSTER_STARTUP_NODES
-    if startup_nodes is None:
-        startup_nodes = [{"host": settings.REDIS_HOST, "port": settings.REDIS_PORT}]
-    redis_client = RetryingStrictRedisCluster(
-        startup_nodes=startup_nodes,
-        socket_keepalive=True,
-        password=settings.REDIS_PASSWORD,
-        # HACK(mattrobenolt): See https://github.com/Grokzen/redis-py-cluster/pull/353
-        max_connections_per_node=True,
-    )
-else:
-    redis_client = StrictRedis(
-        host=settings.REDIS_HOST,
-        port=settings.REDIS_PORT,
-        password=settings.REDIS_PASSWORD,
-        db=settings.REDIS_DB,
-        socket_keepalive=True,
-    )
+def _retry(func: Callable[[], RedisClientType]) -> Callable[[], RedisClientType]:
+    @wraps(func)
+    def wrapper() -> RedisClientType:
+        retry_counter = 3
```

Comment:
nit: it's nice if magic numbers/constants are split out from logic (and even methods).

`MAX_RETRIES=3`
`KNOWN_REDIS_INITIALIZATION_NOT_FINISHED_MESSAGE = "All slots are not"`
`RANDOM_SLEEP_MAX=50`

Additionally, it would be cool if _retry could be parameterized so the annotation is something like

`@_retry(3)`

Reply:
Having the amount of retries in settings also allows us to have a different behavior between self hosted/dev and prod, so please let's do that way.